src/module_wrap.cc

```diff
@@ -980,40 +985,50 @@ Maybe<URL> ResolveExportsTarget(Environment* env,
       Local<Array> target_arr = target.As<Array>();
       const uint32_t length = target_arr->Length();
       if (length == 0) {
-        if (throw_invalid) {
-          ThrowExportsInvalid(env, pkg_subpath, target, pjson_url, base);
-        }
+        ThrowExportsInvalid(env, pkg_subpath, target, pjson_url, base);
         return Nothing<URL>();
       }
       for (uint32_t i = 0; i < length; i++) {
         auto target_item = target_arr->Get(context, i).ToLocalChecked();
-        if (!target_item->IsArray()) {
+        {
+          TryCatchScope try_catch(env);
           Maybe<URL> resolved = ResolveExportsTarget(env, pjson_url,
-              target_item, subpath, pkg_subpath, base, false);
-          if (resolved.IsNothing()) continue;
+              target_item, subpath, pkg_subpath, base);
+          if (resolved.IsNothing()) {
+            CHECK(try_catch.HasCaught() && !try_catch.Exception().IsEmpty());
+            auto e = try_catch.Exception()->ToObject(context).ToLocalChecked();
+            auto code = e->Get(context, env->code_string()).ToLocalChecked();
+            Utf8Value code_utf8(env->isolate(),
+                                code->ToString(context).ToLocalChecked());
+            std::string code_str(*code_utf8, code_utf8.length());
+            if (code_str == "ERR_PACKAGE_PATH_NOT_EXPORTED" ||
+                code_str == "ERR_INVALID_PACKAGE_TARGET") {
```

Comment:
If you’re only constructing `code_str` to compare the string, it’s probably better to compare `code_utf8` directly (i.e. `if (strcmp(*code_utf8, "ERR_PACKAGE_PATH_NOT_EXPORTED") == 0 || ...`)

(All these comments apply below, too)

Reply:
Thanks, included.